ext/session/session.c

```diff
@@ -710,6 +710,23 @@ static PHP_INI_MH(OnUpdateCookieLifetime) /* {{{ */
 }
 /* }}} */
 
+static PHP_INI_MH(OnUpdateCacheExpire)
+{
+	SESSION_CHECK_ACTIVE_STATE;
+	SESSION_CHECK_OUTPUT_STATE;
+
+#ifdef ZEND_ENABLE_ZVAL_LONG64
+	const zend_long maxexpire = ((ZEND_LONG_MAX - INT_MAX) / 60) - 1;
+#else
+	const zend_long maxexpire = ((ZEND_LONG_MAX / 2) / 60) - 1;
+#endif
+	zend_long v = (zend_long)atol(ZSTR_VAL(new_value));
+	if (v < 0 || v > maxexpire) {
+		return SUCCESS;
+	}
```

Comment:
Surely this should be `return FAILURE` ?

Reply:
In fact I applied the same "policy" as for cookie_lifetime as to not disturb things for stable branches but I can t return FAILURE sure.

Reply:
I think this strategy makes sense.

However, the approach appears to be insufficient. For 32bit architectures we have `((ZEND_LONG_MAX / 2) / 60) - 1;`. Asssuming that `time_t` is a signed 32bit, that can still overflow: let's say the current timestamp is `1728992731`; the max value would be `7456539`. Then we do `tv.tv_sec + PS(cache_expire) * 60`, which results in signed overflow. For our Windows 32bit builds, `time_t` is `long` which is a signed 32bit value on LLP64.

It seems to me that we need to do the overflow check when we're actually calculating the time (maybe instead of failing, just clamping the value). Rejecting very large values in the INI modifcation handler makes sense, but doesn't solve all issues; not even for 64bit architectures when running this code in say 15 years.

Reply:
alright thx, I ll get back to it in couple of hours.